core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java

```diff
@@ -6,18 +6,32 @@
  */
 package org.mule.runtime.core.internal.management.stats;
 
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Spliterator;
 import java.util.Spliterators;
 import java.util.function.LongConsumer;
 
 import org.apache.commons.collections.collection.AbstractCollectionDecorator;
 
-class PayloadStatisticsCollection<T> extends AbstractCollectionDecorator {
+class PayloadStatisticsCollection<T> extends AbstractCollectionDecorator implements Externalizable {
 
   private final LongConsumer populator;
 
+  public PayloadStatisticsCollection() {
```

Comment:
would this be needed if there were a kryo serializer for this class?

Reply:
the void constructor is needed for non-kryo scenarios. Kryo also needs all parameters to be initialized in the void constructor.